packages/dd-trace/test/appsec/iast/telemetry/namespaces.spec.js

```diff
@@ -71,24 +72,51 @@ describe('IAST metric namespaces', () => {
   })
 
   it('should merge all kind of metrics in global Namespace as gauges', () => {
-    namespace.count(REQUEST_TAINTED, { tag1: 'test' }).inc(10)
+    namespace.count(REQUEST_TAINTED, ['tag1:test']).inc(10)
     namespace.count(EXECUTED_SINK).inc(1)
 
     const metric = {
       inc: sinon.spy()
     }
-    sinon.stub(globalNamespace, 'count').returns(metric)
+    const count = sinon.stub(Namespace.prototype, 'count').returns(metric)
 
     finalizeRequestNamespace(context, rootSpan)
 
-    expect(globalNamespace.count).to.be.calledTwice
-    expect(globalNamespace.count.firstCall.args).to.be.deep.equal([REQUEST_TAINTED, ['tag1:test']])
+    expect(count).to.be.calledTwice
+    expect(count.firstCall.args).to.be.deep.equal([REQUEST_TAINTED, ['tag1:test']])
     expect(metric.inc).to.be.calledTwice
     expect(metric.inc.firstCall.args[0]).to.equal(10)
 
-    expect(globalNamespace.count.secondCall.args).to.be.deep.equal([EXECUTED_SINK, []])
+    expect(count.secondCall.args).to.be.deep.equal([EXECUTED_SINK, undefined])
     expect(metric.inc.secondCall.args[0]).to.equal(1)
   })
+
+  it('should cache metrics from different request namespaces', () => {
+    const context2 = {}
+    const namespace2 = initRequestNamespace(context2)
+    namespace2.count(REQUEST_TAINTED, { tag1: 'test' }).inc(10)
+
+    finalizeRequestNamespace(context2)
+
+    const context3 = {}
+    const namespace3 = initRequestNamespace(context3)
+    namespace3.count(REQUEST_TAINTED, { tag1: 'test' }).inc(10)
+
+    finalizeRequestNamespace(context3)
+
+    expect(globalNamespace.iastMetrics.size).to.be.eq(1)
+  })
+
+  it('should clear metric and distribution collections and iast metrics cache', () => {
+    namespace.count(REQUEST_TAINTED, ['tag1:test']).inc(10)
+    namespace.distribution('test.distribution', ['tag2:test']).track(10)
+
+    finalizeRequestNamespace(context)
+
+    expect(namespace.iastMetrics.size).to.be.eq(0)
+    expect(namespace.metrics.size).to.be.eq(0)
+    expect(namespace.distributions.size).to.be.eq(0)
```

Comment:
`to.be.equal(` is more used than `eq`
```suggestion
    expect(namespace.iastMetrics.size).to.be.equal(0)
    expect(namespace.metrics.size).to.be.equal(0)
    expect(namespace.distributions.size).to.be.equal(0)
```